src/main/java/com/slack/eithernet/Retries.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2023 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.slack.eithernet
+
+import kotlin.random.Random
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.delay
+
+/**
+ * Retries a [block] of code with exponential backoff.
+ *
+ * @param maxAttempts The maximum number of times to retry the operation. Default is 5.
+ * @param initialDelay The delay before the first retry. Default is 1 second.
+ * @param delayFactor The factor by which the delay should increase after each failed attempt.
+ *   Default is 2.0.
+ * @param maxDelay The maximum delay between retries. Default is [Duration.INFINITE].
```

Comment:
nit: this is a dumb delay. 1 hour is probably better.

Reply:
(Because if you have an outage, you don’t actually ever want to wait an unbounded amount of time before reconnecting.)

Reply:
Fair enough! [13439d0](https://github.com/slackhq/EitherNet/pull/60/commits/13439d0d33f0382a2587f76c8745a2838d2504b5)